File: LeetCode/0007.go
package solutions

import "math"

func reverse(x int) int {
	sign := 1

	// 處理正負
	if x < 0 {
		sign = -1
		x *= -1
	}

	// 數字反轉
	res := 0

	for x > 0 {
		res = res*10 + x%10
		x = x / 10
	}

	// 還原正負
	res *= sign

	// 處理 overflow
	if res > math.MaxInt32 || res < math.MinInt32 {
		res = 0
	}

	return res
}

File: LeetCode/0007.py
# O(log x) O(1)
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x < 0 else 1
        res = int(str(abs(x))[::-1]) * sign
        return res if -2**31 <= res <= 2**31 -1 else 0
